js/src/setup-mc/setup-stepper/setup-free-listings/useAutoSaveShippingRatesEffect.js

```diff
@@ -1,15 +1,37 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import { useAppDispatch } from '.~/data';
 import useIsEqualRefValue from '.~/hooks/useIsEqualRefValue';
 import useDebouncedCallbackEffect from '.~/hooks/useDebouncedCallbackEffect';
+import useSaveShippingRates from '.~/hooks/useSaveShippingRates';
 
 const useAutoSaveShippingRatesEffect = ( shippingRates ) => {
-	const { saveShippingRates } = useAppDispatch();
+	const { saveShippingRates } = useSaveShippingRates();
 	const shippingRatesRefValue = useIsEqualRefValue( shippingRates );
 
-	useDebouncedCallbackEffect( shippingRatesRefValue, saveShippingRates );
+	/**
+	 * A `saveShippingRates` callback that catches error and do nothing.
+	 * We don't want to show error messages for this auto save feature,
+	 * and want it to fail silently in the background.
+	 */
+	const saveShippingRatesCallback = useCallback(
+		async ( value ) => {
+			try {
+				await saveShippingRates( value );
+			} catch ( error ) {}
+		},
+		[ saveShippingRates ]
+	);
```

Comment:
💅 
The callback reference changes don't affect the use of `useDebouncedCallback`. The `useCallback` could be omitted.

Reply:
Ahh right, I forgot that is already handled by [`useDebouncedCallback`](https://github.com/xnimorz/use-debounce#debounced-callbacks). Removed in [`e9f512b` (#1313)](https://github.com/woocommerce/google-listings-and-ads/pull/1313/commits/e9f512bcb99d7baa61ab1760504b8919d4059229). 👍